config/initializers/makara.rb

```diff
@@ -0,0 +1,136 @@
+# don't indicate cached queries are being run against the primary DB
+module MakaraLoggingSubscriber
+  # See https://github.com/instacart/makara/blob/e45ba090fce998dad9e9a2759426f4695009cfae/lib/makara/logging/subscriber.rb#L23
+  def current_wrapper_name(event)
+    return nil if event.payload[:cached]
+    super(event)
+  end
+end
+
+# apply the above fix for cached query logging
+ActiveRecord::LogSubscriber.log_subscribers.each do |subscriber|
+  subscriber.extend ::MakaraLoggingSubscriber
+end
+
+# fix for makara DB name logging for Rails 6
+module Makara
+  module Logging
+    module Subscriber
+      protected
+
+      def current_wrapper_name(event)
+        adapter = event.payload[:connection]
+
+        return nil unless adapter
+        return nil unless adapter.respond_to?(:_makara_name)
+
+        "[#{adapter._makara_name}]"
+      end
+    end
+  end
+end
+
+# ensure MakaraAbstractAdapter is defined even for environments not using makara
+module ActiveRecord
+  module ConnectionAdapters
+    class MakaraAbstractAdapter < ::Makara::Proxy
+
+      SQL_PRIMARY_MATCHERS = [] unless defined?( SQL_PRIMARY_MATCHERS )
+      CUSTOM_SQL_PRIMARY_MATCHERS = SQL_PRIMARY_MATCHERS + [/delayed_jobs/i].map(&:freeze).freeze
+      SQL_SKIP_STICKINESS_MATCHERS = [] unless defined?( SQL_SKIP_STICKINESS_MATCHERS )
+      CUSTOM_SQL_SKIP_STICKINESS_MATCHERS  = SQL_SKIP_STICKINESS_MATCHERS + [/FROM "sessions"/i].map(&:freeze).freeze
+
+      def sql_primary_matchers
+        CUSTOM_SQL_PRIMARY_MATCHERS
+      end
+
+      def sql_skip_stickiness_matchers
+        CUSTOM_SQL_SKIP_STICKINESS_MATCHERS
+      end
```

Comment:
This makes sure the module is defined, but what are these methods doing?

Reply:
`sql_skip_stickiness_matchers` is defined in the Makara library, and I'm overriding it here. These are patterns where, if the current query matches any of the patterns, it'll automatically ignore stickiness. I wanted to extend the library defaults w/o repeating the default values, so I'm extending them with the CUSTOM_* constants, and overriding the methods to use the extended matchers. I can add some more comments in this file